src/window/index.ts

```diff
@@ -125,6 +126,53 @@ export async function openFirstRun() {
   }));
 }
 
+/**
+ * Open the error message window as a modal window.
+ */
+export async function openKubernetesErrorMessageWindow(titlePart: string, mainMessage: string, lastCommand: string, lastCommandComment: string, logLines: string[]) {
+  const webRoot = getWebRoot();
+  // We use hash mode for the router, so `index.html#FirstRun` loads
+  // src/pages/FirstRun.vue.
+  const options: Electron.BrowserWindowConstructorOptions = {
+    width:           800,
+    height:          494,
+    minWidth:        800,
+    minHeight:       494,
+    autoHideMenuBar: !app.isPackaged,
+    show:            false,
+    alwaysOnTop:     true,
+    closable:        true,
+    maximizable:     false,
+    minimizable:     false,
+    modal:           true,
+    webPreferences:  {
+      devTools:           !app.isPackaged,
+      nodeIntegration:    true,
+      contextIsolation:   false,
+    },
+  };
+  const preferencesWindow = BrowserWindow.fromId(windowMapping['preferences']);
+
+  if (preferencesWindow) {
+    options.parent = preferencesWindow;
+  }
```

Comment:
Sure. Note that I had to do `BrowserWindow.fromId(windowMapping['preferences']) ?? undefined` because `null` actually isn't accepted by the type `Electron.BrowserWindowConstructorOptions`.